doc/setup.md

```diff
@@ -1,6 +1,21 @@
-## Setup
+# Setup
 
-To use N-API in a native module:
+## Prerequisites
+
+Before starting to use **N-API** you need to assure you have the following 
+prerequisites:
+
+* **Node.JS** see: [Installing Node.js](https://nodejs.org/)
+
+* **Node.js native addon build tool** 
+
+  - **[node-gyp](node-gyp.md)**
+
+  - **[cmake-js](cmake-js.md)**
```

Comment:
How widely used is cmake-js ? Just wondering if we should be calling it out here compared to node-gyp which is used for most modules.

Reply:
I we consider the npm statistics yes node-gyp is the de-facto build tool for the node.js addon, but based on this issue https://github.com/nodejs/node-addon-api/issues/83 I thinked to insert some reference on documentation, maybe we can remove from **setup.md**  and maintain only on **readme.md**

Reply:
ok that sounds good.

Reply:
Removed c-make-js from setup.md